metricbeat/module/system/filesystem/filesystem.go

```diff
@@ -0,0 +1,56 @@
+package filesystem
+
+import (
+	"github.com/elastic/beats/metricbeat/helper"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/topbeat/system"
+)
+
+func init() {
+	helper.Registry.AddMetricSeter("system", "filesystem", New)
+}
+
+// New creates new instance of MetricSeter
+func New() helper.MetricSeter {
+	return &MetricSeter{}
+}
+
+type MetricSeter struct{}
+
+func (m *MetricSeter) Setup(ms *helper.MetricSet) error {
+	return nil
+}
+
+func (m *MetricSeter) Fetch(ms *helper.MetricSet, host string) (event common.MapStr, err error) {
+
+	fss, err := system.GetFileSystemList()
+	if err != nil {
+		logp.Warn("Getting filesystem list: %v", err)
+		return nil, err
+	}
+
+	event = common.MapStr{}
+
+	for _, fs := range fss {
+		fsStat, err := system.GetFileSystemStat(fs)
+		if err != nil {
+			logp.Debug("topbeat", "Skip filesystem %d: %v", fsStat, err)
+			continue
+		}
+		system.AddFileSystemUsedPercentage(fsStat)
+
+		fsEvent := common.MapStr{
+			fsStat.DevName: system.GetFilesystemEvent(fsStat),
```

Comment:
I'm afraid arrays are also not a good option when it comes to visualizing in Kibana, see for e.g. https://github.com/elastic/kibana/issues/998. Besides the visualization aspect, having ephemeral fields like PID is quite space inefficient, because it tends to create a lot of sparse doc values.

I was thinking the metric name will be something like `filesystem.size` and the device name would be a label, just like the host, for example. IMO putting the device name or PID into the metric name sends us back to the Graphite ways where this is the only way to put metadata in.